### DISCUSSION TITLE
Preventing examples in loops

### DISCUSSION THREAD

**Discussion by BigBigDoudou** [rubocop-hq/rubocop-rspec#1782] [Ideas]
Hey! 👋 

In many specs, I see examples written dynamically inside loops:

```ruby
describe 'Enums translations' do
  enums.each do |enum|
    it "enum #{enum} has a translation" do
      # ...
    end
  end
end
```

However, it poses certain problems:
- it is difficult to maintain overall
- failures are difficult to read, as RSpec displays the iteration (`'/path_to_spec.rb[1:1:42]'`)
- some configurations won't work as expected, e.g. `eager_load` works inside a block, not outside, so doing `ApplicationRecord.descendants.each { it { ... } }` will not work because the models are not loaded.

The idea would therefore be to have a cop that prevents examples in loops, so the specification should be replaced by:

```ruby
describe 'Enums translations' do
  it 'translates all enums', :aggregate_failures do
    enums.each do |enum|
      expect(...).to eq(...)
    end
  end
end
```

Note that I use `aggregate_failures` to ensure that all elements of the enumerable are tested, as they were before.

WDYT?

**Comment by pirj** [comment]
It will certainly be a good addition.

Please note that we have a cop that would take it further and suggest `expect(enum).to all eq(…)`

Would you like to hack on it?
